bazel/bazel.go

```diff
@@ -23,13 +23,66 @@ import (
 	"io"
 	"os"
 	"os/exec"
+	"path/filepath"
 	"strings"
 
 	blaze_query "github.com/bazelbuild/bazel-watcher/third_party/bazel/master/src/main/protobuf"
 	"github.com/golang/protobuf/proto"
 )
 
-var bazelPath = flag.String("bazel_path", "bazel", "Path to the bazel binary to use for actions")
+var bazelPathFlag = flag.String("bazel_path", "", "Path to the bazel binary to use for actions")
+
+// bazelNpmPath looks up a relative path to a binary from @bazel/bazel
+// This is used as an alternate resolution when no bazel binary is in the $PATH
+// When running from the @bazel/ibazel npm package, our binary is
+// /DIR/node_modules/@bazel/ibazel/bin/$ARCH/ibazel
+// We can find bazel in
+// /DIR/node_modules/@bazel/bazel-$ARCH/bazel-0.28.0-linux-x86_64
+func bazelNpmPath(ibazelBinPath string) (string, error) {
+	s := strings.Split(ibazelBinPath, "/")
+	for i := 0; i + 4 < len(s); i++ {
+		nm, scope, pkg, dir, bin := s[i], s[i+1], s[i+2], s[i+3], s[i+4]
+		if nm == "node_modules" && scope == "@bazel" && pkg == "ibazel" && dir == "bin" {
```

Comment:
Thanks for doing this! The conditional here is, indeed, very simple to reason through.